jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpField.java

```diff
@@ -38,50 +42,85 @@ public class HttpField
+     * @param value the field value; if {@code null}, the empty string will be used
+     */
     public HttpField(HttpHeader header, String name, String value)
     {
         _header = header;
         if (_header != null && name == null)
             _name = _header.asString();
         else
             _name = Objects.requireNonNull(name, "name");
-        _value = value;
+        _value = value != null ? value : "";
     }
 
+    /**
+     * <p>Creates a new {@link HttpField} with the given {@link HttpHeader}
+     * and value string.</p>
+     *
+     * @param header the non-{@code null} {@link HttpHeader} referencing a well-known HTTP header name
+     * @param value the field value; if {@code null}, the empty string will be used
+     */
     public HttpField(HttpHeader header, String value)
     {
         this(header, header.asString(), value);
     }
 
+    /**
+     * <p>Creates a new {@link HttpField} with the given {@link HttpHeader}
+     * and value.</p>
+     *
+     * @param header the non-{@code null} {@link HttpHeader} referencing a well-known HTTP header name
+     * @param value the field value; if {@code null}, the empty string will be used
+     */
     public HttpField(HttpHeader header, HttpHeaderValue value)
     {
-        this(header, header.asString(), value.asString());
+        this(header, header.asString(), value != null ? value.asString() : null);
     }
 
+    /**
+     * <p>Creates a new {@link HttpField} with the given name string
+     * and value string.</p>
+     *
+     * @param name the non-{@code null} field name
+     * @param value the field value; if {@code null}, the empty string will be used
+     */
     public HttpField(String name, String value)
     {
         this(HttpHeader.CACHE.get(name), name, value);
     }
 
     /**
-     * Get field value parameters. Some field values can have parameters. This method separates the
-     * value from the parameters and optionally populates a map with the parameters. For example:
-     *
-     * <PRE>
+     * <p>Returns the field value and its parameters.</p>
+     * <p>A field value may have parameters, typically separated by {@code ;}, for example</p>
+     * <pre>{@code
+     * Content-Type: text/plain; charset=UTF-8
+     * Accept: text/html, text/plain; q=0.5
+     * }</pre>
      *
-     * FieldName : Value ; param1=val1 ; param2=val2
-     *
-     * </PRE>
-     *
-     * @param valueParams The Field value, possibly with parameters.
-     * @param parameters A map to populate with the parameters, or null
-     * @return The value.
+     * @param valueParams the field value, possibly with parameters
+     * @param parameters An output map to populate with the parameters,
+     * or {@code null} to strip the parameters
+     * @return the field value without parameters
+     * @see #stripParameters(String)
      */
     public static String getValueParameters(String valueParams, Map<String, String> parameters)
```

Comment:
This method also has a code smell. It's uncommon for getter method to update the value of its parameters. It seems to be doing two different things (which implies it should be separated into different methods). Consider:

```
Map<String, String> getValueParameters();
String getValueWithoutParameters();
```

Alternatively:

```
record HeaderValue(String value, Map<String, String> parameters)
{}

static HeaderValue parseHeaderValue(String value);
```

Reply:
Won't be efficient, as we would parse the value twice.
There are other examples in the JDK of parameters that are used as "extra" return value, see `AtomicMarkableReference.get()` for example.

Reply:
You wouldn't have to parse the value twice if you use the record approach. If you can use records, just create a normal class.

Reply:
But many usages do not expect nor use parameters, so they just pass in null for the map.   Returning a record would always create a record object and a populated map, even if not used.

Reply:
I think you misunderstood what I meant, so I'll give this one last try. If you still don't like the proposal then I'll drop it.

```
record ValueWithParameters(String value, Map<String, String> parameters)
{}

ValueWithParameters getValueWithParameters();
String getValueWithoutParameters();
```

* If the caller wants parameters, they invoke the first method.
* If the caller does not want parameters, they invoke the second method.

@sbordet I don't understand why you said this would require parsing the value twice.
@gregw If a user does not care about the parameters, they could invoke the second method. No record would get constructed.